### PR TITLE
Peep2egg second iteration

### PR DESCRIPTION
**Pull request by Jacarte**
CC @fitzgen 

Next iteration for the migration of peephole mutators to egg.

### CODE REVIEW

File: crates/wasm-mutate/src/mutators/peephole/eggsy/lang.rs
Comment (fitzgen) [inline crates/wasm-mutate/src/mutators/peephole/eggsy/lang.rs:31]: How is this different from symbol, which is also just some opaque value we can manipulate but can't peek at actual value? Also, it seems like there can only be a single `undef` in an egg term as well, since we don't have a way of differentiating them, which is pretty restricting.

File: crates/wasm-mutate/src/mutators/peephole/eggsy/mod.rs
Comment (fitzgen) [inline crates/wasm-mutate/src/mutators/peephole/eggsy/mod.rs]: These can probably be `debug_assert_eq!` as they are "internal" implementation assertions and not something we are asserting about user/some-other-crate's input.

**Review by fitzgen — APPROVED** [review]
Looks good to me; will hold off on merging until after #352 merges.

**Review by fitzgen — APPROVED** [review]
Overall, looking good. I have a few questions/comments, but we can address these in follow up PRs.

File: crates/wasm-mutate/src/mutators/peephole/eggsy/lang.rs
Comment (Jacarte) [reply crates/wasm-mutate/src/mutators/peephole/eggsy/lang.rs:30]: This symbol is used right now to determine if the DFG is determined, if the generated DFG written as an eterm has the undef node, then the operand for some operator comes from a different basic block.

File: crates/wasm-mutate/src/mutators/peephole/eggsy/mod.rs
Comment (Jacarte) [reply crates/wasm-mutate/src/mutators/peephole/eggsy/mod.rs:144]: Fixed

**Pull request merged**
